daemon/logger/jsonfilelog/jsonlog/jsonlog.go

```diff
@@ -21,5 +21,7 @@ func (jl *JSONLog) Reset() {
 	jl.Log = ""
 	jl.Stream = ""
 	jl.Created = time.Time{}
-	jl.Attrs = make(map[string]string)
+	for k := range jl.Attrs {
+		delete(jl.Attrs, k)
+	}
```

Comment:
Is the explicit delete better than assigning an empty map? From what I understood it's the same thing.

Reply:
In one case it has to allocate, in the other not.
Also this pattern for deleting the map is optimized by the compiler.

Reply:
It's worth noting that right now I believe map elements that are deleted won't actually be GCed until the map itself is GCed, see golang/go#20135

Reply:
so would the old approach be better? should we manually trigger a garbage collect?

Reply:
Realistically, this is not going to change in size. We reset it because it's proper b/c things *could* change, but in reality it is not changing.

Reply:
Without doing some profiling I'm really not sure. How often are these reset vs. created/destroyed? Maybe the higher memory footprint is worth it if we avoid an expensive allocation, or maybe we'd rather take one more allocation to avoid the higher memory footprint.

Reply:
I do not think there is any higher memory footprint because these attrs actually never change.